### ISSUE TITLE
SSL certificate with unsupported SAN name type

### ISSUE THREAD

**Issue by p4yne** [mail-in-a-box/mailinabox#481]
Error Message on the CLI after logging in and requesting the status checks page:
```bash
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.4/multiprocessing/pool.py", line 420, in _handle_results
    task = get()
  File "/usr/lib/python3.4/multiprocessing/connection.py", line 251, in recv
    return ForkingPickler.loads(buf.getbuffer())
TypeError: __init__() missing 1 required positional argument: 'type'
```
The request the times out in the browser. And not a single request is then handled by the application any more, requests time out every time.

As this error is rather uninformative, I poked around a bit more and when requesting the ssl certificates page directly after restarting the daemon, I get the following error:

```bash
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3/dist-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3/dist-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/<userhere>/mailinabox/management/daemon.py", line 49, in newview
    return viewfunc(*args, **kwargs)
  File "/home/<userhere>/mailinabox/management/daemon.py", line 333, in web_get_domains
    return json_response(get_web_domains_info(env))
  File "/home/<userhere>/mailinabox/management/web_update.py", line 328, in get_web_domains_info
    for domain in get_web_domains(env)
  File "/home/<userhere>/mailinabox/management/web_update.py", line 328, in <listcomp>
    for domain in get_web_domains(env)
  File "/home/<userhere>/mailinabox/management/web_update.py", line 308, in check_cert
    cert_status, cert_status_details = check_certificate(domain, ssl_certificate, ssl_key)
  File "/home/<userhere>/mailinabox/management/status_checks.py", line 652, in check_certificate
    sans = cert.extensions.get_extension_for_oid(OID_SUBJECT_ALTERNATIVE_NAME).value.get_values_for_type(DNSName)
  File "/usr/local/lib/python3.4/dist-packages/cryptography/hazmat/backends/openssl/x509.py", line 310, in extensions
    return _CERTIFICATE_EXTENSION_PARSER.parse(self._backend, self._x509)
  File "/usr/local/lib/python3.4/dist-packages/cryptography/hazmat/backends/openssl/x509.py", line 197, in parse
    value = handler(backend, ext)
  File "/usr/local/lib/python3.4/dist-packages/cryptography/hazmat/backends/openssl/x509.py", line 530, in _decode_subject_alt_name
    _decode_general_names_extension(backend, ext)
  File "/usr/local/lib/python3.4/dist-packages/cryptography/hazmat/backends/openssl/x509.py", line 524, in _decode_general_names_extension
    general_names = _decode_general_names(backend, gns)
  File "/usr/local/lib/python3.4/dist-packages/cryptography/hazmat/backends/openssl/x509.py", line 77, in _decode_general_names
    names.append(_decode_general_name(backend, gn))
  File "/usr/local/lib/python3.4/dist-packages/cryptography/hazmat/backends/openssl/x509.py", line 162, in _decode_general_name
    gn.type
cryptography.x509.UnsupportedGeneralNameType: otherName is not a supported type
```

Which is an error thrown by [pyca/cryptography](https://github.com/pyca/cryptography/blob/master/src/cryptography/hazmat/backends/openssl/x509.py), I guess because of my certificate containing the following:

```bash
 X509v3 Subject Alternative Name: 
                DNS:<domainnamehere>, othername:<unsupported>
```
Things changed in the above output "domainnamehere" and "userhere".

Before the switch from openssl to pyca everything was working fine, at least as far as I can tell ;-) !

**Comment by JoshData** [comment]
I'm working on an upstream fix for this. See the PR above.

**Comment by p4yne** [comment]
Yeah, I thought that this might be an upstream issue, but wasn't entirely sure. But thank you for working on a fix/upgrade to the pyca project!

**Comment by JoshData** [comment]
I fixed the upstream problem, so you can do this to get around the issue:

    sudo pip3 install git+https://github.com/pyca/cryptography
    sudo service mailinabox restart

**Comment by p4yne** [comment]
Thank you again, I can confirm this solves the issue.
Keep up the awesome work!

**Comment by JoshData** [comment]
Just noting that the cryptography library released 1.0 a few weeks ago with this fix so this should be working in the latest Mail-in-a-Box now.
